lib/datadog/core/configuration/config_helper.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require_relative 'assets/supported_configurations'
+require_relative '../logger'
+require 'json'
+
+module Datadog
+  module Core
+    module Configuration
+      module ConfigHelper
+        # Load the supported configurations from JSON file. We will do this AOT.
+        SUPPORTED_CONFIGURATIONS = Assets::SUPPORTED_CONFIG_DATA[:supportedConfigurations] || {}
+        private_constant :SUPPORTED_CONFIGURATIONS
+
+        ALIASES = Assets::SUPPORTED_CONFIG_DATA[:aliases] || {}
+        private_constant :ALIASES
+
+        DEPRECATIONS = Assets::SUPPORTED_CONFIG_DATA[:deprecations] || {}
+        private_constant :DEPRECATIONS
+
+        ALIAS_TO_CANONICAL = Assets::ALIAS_TO_CANONICAL || {}
+        private_constant :ALIAS_TO_CANONICAL
+
+        # Returns the environment variable, if it's supported or a non Datadog
+        # configuration. Otherwise, it throws an error.
+        #
+        # @param name [String] Environment variable name
+        # @return [String, nil] The environment variable value
+        # @raise [RuntimeError] if the configuration is not supported
+        def get_environment_variable(name, env_vars: ENV, source: 'environment')
+          # Log deprecations once for environment, fleet config file and local config file.
+          log_deprecations(env_vars, source)
```

Comment:
we will log deprecations here for all deprecated environment variables (if they are set on the machine), even if some environment variable is not being accessed in our code. Do we really want that?